src/lambda/commands/deploySamApplication.ts

```diff
@@ -40,7 +43,17 @@ export async function deploySamApplication({
         const buildTemplatePath = path.join(buildDestination, 'template.yaml')
         const outputTemplatePath = path.join(tempFolder, 'template.yaml')
         let stage: 'starting up' | 'building' | 'packaging' | 'deploying' = 'starting up'
+
+        const settingsConfiguration = new DefaultSettingsConfiguration(extensionSettingsPrefix)
```

Comment:
We shouldn't explicitly create a DefaultSettingsConfiguration here, as it prevents mocking. Instead:

1. Add a second `context` parameter.
2. Move `invoker` and `regionProvider` (not sure about `outputChannel`) to the `context` parameter.
3. Add `settingsConfiguration` to the context parameter with a default value of `new DefaultSettingsConfiguration(ExtensionSettingsPrefix)`.

Reply:
Why not throw it into `restParams`?

Reply:
✔️ Removed this and added a context. It is not included with the rest of the parameters (open to changing that)